lib/crashlogger.ts

```diff
@@ -8,46 +8,39 @@
  * @license MIT
  */
 
-'use strict';
-
-const fs = require('fs');
-const path = require('path');
+import * as fs from 'fs';
+import * as path from 'path';
 
 const CRASH_EMAIL_THROTTLE = 5 * 60 * 1000; // 5 minutes
 const LOCKDOWN_PERIOD = 30 * 60 * 1000; // 30 minutes
 
 const logPath = path.resolve(__dirname, '../logs/errors.txt');
 let lastCrashLog = 0;
-/** @type {any} */
-let transport;
+let transport: any;
 
 /**
  * Logs when a crash happens to console, then e-mails those who are configured
  * to receive them.
- *
- * @param {Error | string} err
- * @param {string} description
- * @param {?Object} [data = null]
- * @return {?string}
  */
-module.exports = function crashlogger(err, description, data = null) {
+export = function crashlogger(err: Error | string, description: string, data: object | null = null): string | null {
 	const datenow = Date.now();
 
 	let stack = typeof err === 'string' ? err : err.stack;
 	if (data) {
 		stack += `\n\nAdditional information:\n`;
 		for (let k in data) {
+			// @ts-ignore
 			stack += `  ${k} = ${data[k]}\n`;
 		}
 	}
 
 	console.error(`\nCRASH: ${stack}\n`);
-	let out = fs.createWriteStream(logPath, {'flags': 'a'});
+	let out = fs.createWriteStream(logPath, {flags: 'a'});
 	out.on('open', () => {
 		out.write(`\n${stack}\n`);
 		out.end();
-	}).on('error', /** @param {Error} err */ err => {
-		console.error(`\nSUBCRASH: ${err.stack}\n`);
+	}).on('error', (e: Error) => {
```

Comment:
I'd actually prefer to consistently use `err` rather than `e` for errors; I've been dealing with some conflicts on client for that because `e` usually means `event` there.

Reply:
Oh, you're avoiding shadowing. Maybe call the outer one `error`.